axelor-account/src/main/java/com/axelor/apps/account/service/notebills/NoteBillsCreateServiceImpl.java

```diff
@@ -0,0 +1,68 @@
+package com.axelor.apps.account.service.notebills;
+
+import com.axelor.apps.account.db.AccountingBatch;
+import com.axelor.apps.account.db.NoteBills;
+import com.axelor.apps.account.db.repo.NoteBillsRepository;
+import com.axelor.apps.account.exception.IExceptionMessage;
+import com.axelor.apps.base.db.Batch;
+import com.axelor.apps.base.db.Company;
+import com.axelor.apps.base.db.Partner;
+import com.axelor.apps.base.db.repo.SequenceRepository;
+import com.axelor.apps.base.service.administration.SequenceService;
+import com.axelor.exception.AxelorException;
+import com.axelor.exception.db.repo.TraceBackRepository;
+import com.axelor.i18n.I18n;
+import com.google.inject.Inject;
+import com.google.inject.persist.Transactional;
+import java.util.Objects;
+
+public class NoteBillsCreateServiceImpl implements NoteBillsCreateService {
+
+  protected NoteBillsRepository noteBillsRepository;
+  protected SequenceService sequenceService;
+
+  @Inject
+  public NoteBillsCreateServiceImpl(
+      NoteBillsRepository noteBillsRepository, SequenceService sequenceService) {
+    this.noteBillsRepository = noteBillsRepository;
+    this.sequenceService = sequenceService;
+  }
+
+  @Override
+  @Transactional
```

Comment:
Rollback should be managed here. https://github.com/google/guice/wiki/Transactions#responding-to-exceptions